app/components/primer/alpha/x_banner.ts

```diff
@@ -1,18 +1,26 @@
 import {controller, target} from '@github/catalyst'
 
+declare global {
+  interface HTMLElementEventMap {
+    dismiss: CustomEvent<void>
+  }
+}
+
 @controller
 class XBannerElement extends HTMLElement {
   @target titleText: HTMLElement
 
   dismiss() {
-    const parentElement = this.parentElement
-    if (!parentElement) return
-
     if (this.#dismissScheme === 'remove') {
+      const parentElement = this.parentElement
+      if (!parentElement) return
+
       parentElement.removeChild(this)
     } else {
       this.hide()
     }
+
+    this.dispatchEvent(new CustomEvent('dismiss'))
```

Comment:
```suggestion
    this.dispatchEvent(new CustomEvent('banner:dismiss'))
```

Should this be more specific? or not necessary

Reply:
I thought about that, but I wasn't sure how to add a global type for it 🤔 Let me do a bit more research.

Reply:
Ok turns out it was really easy, so I made the change 👍